pilot/pkg/bootstrap/server.go

```diff
@@ -1073,18 +1081,25 @@ func (s *Server) maybeCreateCA(caOpts *CAOptions) error {
 }
 
 // startCA starts the CA server if configured.
-func (s *Server) startCA(caOpts *CAOptions) {
-	if s.CA != nil {
-		s.addStartFunc(func(stop <-chan struct{}) error {
-			log.Infof("staring CA")
-			grpcServer := s.secureGrpcServer
-			if s.secureGrpcServer == nil {
-				grpcServer = s.grpcServer
-			}
-			s.RunCA(grpcServer, s.CA, caOpts)
-			return nil
-		})
+func (s *Server) startCA(caOpts *caOptions) {
+	if s.CA == nil && s.RA == nil {
+		return
 	}
+	s.addStartFunc(func(stop <-chan struct{}) error {
+		grpcServer := s.secureGrpcServer
+		if s.secureGrpcServer == nil {
+			grpcServer = s.grpcServer
+		}
+		// TODO: Change logic to start the RA Server
+		if s.RA != nil {
+			log.Infof("Starting RA")
+			s.RunCA(grpcServer, s.RA, caOpts)
+		} else if s.CA != nil {
```

Comment:
nit: I guess you don't need to check s.CA != nil since you already verified it at line 1085.

Reply:
Just wanted to make that chunk of code more readable.